src/core/options-watcher.h

```diff
@@ -0,0 +1,55 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+#pragma once
+
+
+#include <librealsense2/h/rs_option.h>
+#include <src/core/option-interface.h>
+
+#include <rsutils/signal.h>
+#include <rsutils/concurrency/concurrency.h>
+
+#include <map>
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <thread>
+
+
+namespace librealsense {
+
+
+// Automatically updates (queries) registered options values according to set time intervals
+// Can notify if a value have changed through a callback subscription.
+class options_watcher
+{
+public:
+    options_watcher( std::chrono::milliseconds update_interval = std::chrono::milliseconds( 1000 ) );
+    ~options_watcher();
+
+    // Register option to monitor value updates. Can set if update from HW (query) is necessary.
```

Comment:
> Can set if update from HW (query) is necessary.

I think I understand the intent, but I don't see how this is good with a default argument, especially when a `false` means to skip it (line 106 in the cpp)...

I think this is not needed right now. For DDS, this mechanism needs to be overridable. I was going to suggest it before but I think it can be added later on.

Reply:
It was a preparation to the future, when all options will be registered but we will not want to make a transaction for each of them.
I will remove now and we will add again if needed in the future.